plugins/base/src/main/kotlin/DokkaBase.kt

```diff
@@ -191,6 +191,12 @@ class DokkaBase : DokkaPlugin() {
         htmlPreprocessors with RootCreator applyIf { !delayTemplateSubstitution }
     }
 
+    val defaultSamplesTransformer by extending {
+        CoreExtensions.pageTransformer providing ::DefaultSamplesTransformer order {
+            before(pageMerger)
```

Comment:
Could you please remove the `TODO` that I left about this above? This resolves it 

https://github.com/Kotlin/dokka/pull/3102/files#diff-50d822fb0953e3df82e04eb35a5362a37a47adf1390bd2c9fadf97f7f8a6e127R152